gnucash/report/report-core.scm

```diff
@@ -748,7 +749,10 @@ not found.")))
 ;; where captured-error is the error string.
 (define (gnc:render-report report)
   (define (get-report) (gnc:report-render-html report #t))
-  (gnc:apply-with-error-handling get-report '()))
+  (call-with-time-limit
+   10
+   (lambda () (gnc:apply-with-error-handling get-report '()))
+   (lambda () (list #f "Report exceeds time limit. Aborted."))))
```

Comment:
Perhaps inform the user about the value of the time limit?

Reply:
Does this get displayed in the html tab? If not, something should. If so it should be a bit less robotic and marked for translation.